ansible/roles/etcd/tasks/install.yml

```diff
@@ -0,0 +1,18 @@
+---
+- name: Install etcd
+  action: "{{ ansible_pkg_mgr }}"
+  args:
+        name: etcd
+        state: latest
+  notify:
+    - restart etcd
+  when: not is_atomic and ansible_distribution != "Ubuntu" and not is_coreos
+
+- name: Force etcd_source_type to github when packages are not available
+  set_fact:
+    etcd_source_type: "github-release"
```

Comment:
What about to generalize the concept and introduce ``package_source_type``? It could be set in ``pre-ansible`` role. Of course, ``etcd_source_type`` could be still defined to provide higher refinement.